src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
It seems that it is not easy to solve this. Some more changes in determining the appropriate `SolrTemplate`are required. If your solution is appropriate to current design (and maybe API standards) it is maybe the right way. I think i made my concerns clear. Sorry for not providing code on this. But i'm currently really busy. I put it on my list.